duckduckgo/macos-browser
# DBP invite code UI

Task/Issue URL: https://app.asana.com/0/1199230911884351/1204125341274531/f
Tech Design URL:
CC:

**Description**:
Adds the DBP invite code UI to get to the DBP interface. 
Also makes the NP interface use the new slightly more generic interface for the invite code UI.

Also, make sure this PR is merging into the right branch

**Steps to test this PR**:
1. Attempt to open DBP, you should get an invite dialog popup
1. Test when using the cancel button the correct tab closes, and the previously selected tab is the one that becomes selected.
1. Test when typing the wrong code/nonsense you see an error message
1. Test typing the correct code brings up the success dialog. Then test that closing this brings up the DBP interface
1. With the app still open, close the DBP interface, and try accessing it again to make sure you don't have to use an invite code again.
1. Close the app and open it, and again try accessing it again to make sure you don't have to use an invite code again.
1. Test the network protection invite flow still works: https://app.asana.com/0/0/1205174820992574/f



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
